### PR TITLE
All custom names in Character Settings can be left blank to default or randomise them

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

Allows all custom names in Character Settings to be reverted to either the default name or a random one, instead of only cyborg and bible names.

Allows picking a random religion in Character Settings. Changing your religion updates your bible and deity if applicable. It uses the religions and bible names that were already in chaplain spawning code. I've added deities to most of them, to save players having to figure them out themselves. I wasn't familiar with a lot of these religions and references so let me know if I've gotten something wrong.

## Why It's Good For The Game

If someone picks a name they don't like and are too unimaginative to come up with a better one they can revert it to default or randomise it.

If they can't think of a good religion, they can receive a random one. 

## Changelog
:cl: cacogen
qol: All custom names in Character Settings can now either be reset to the default or randomised by leaving the text box blank
qol: You can now pick a random religion in Character Settings by leaving the text box blank. Setting a religion will also set the bible name and deity if applicable
/:cl:

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -2001,3 +2001,91 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 			return
 		else
 			custom_names[name_id] = sanitized_name
+
+	if(name_id == "religion")
+		update_bible_and_deity_name(custom_names[name_id])
+
+/datum/preferences/proc/update_bible_and_deity_name(religion)
+	switch(lowertext(religion))
```
Comment (cacogen) [inline code/modules/client/preferences.dm:2009]: these should probably be in a .json with the religion names but who knows how to do that

File: strings/names/religion_meme.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+poo
+badmin
+shitmin
+deadmin
+cock
+cocks
+meme
+memes
+weeaboo
+kawaii
```
Comment (Rohesie) [inline strings/names/religion_meme.txt:16]: Let's not.

Comment (cacogen) [reply strings/names/religion_meme.txt:16]: I am but the messenger

Comment (Rohesie) [reply strings/names/religion_meme.txt:16]: Remove this crap. There's enough creativity in the playerbase without the game needing to add this kind of suggestions.

**Review by Rohesie — CHANGES_REQUESTED** [review]
Remove the meme religion names. The playerbase has enough creativity to produce these on their own, no need for the game to suggest them.

File: code/__HELPERS/unsorted.dm [hunk 3]
```diff
@@ -1521,3 +1521,6 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 	new /obj/effect/pod_landingzone(landing_location, pod)
 	return pod
 
+/proc/random_religion_name()
+	. = pick(GLOB.religion_names)
+
```
Comment (Rohesie) [inline code/__HELPERS/unsorted.dm:1526]:
```suggestion
```

File: code/modules/client/preferences.dm [hunk 4]
```diff
@@ -1976,7 +1976,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 		if("mime")
 			return pick(GLOB.mime_names)
 		if("religion")
-			return DEFAULT_RELIGION
+			return random_religion_name()
```
Comment (Rohesie) [inline code/modules/client/preferences.dm:1979]:
```suggestion
			return pick(GLOB.religion_names)
```
No need for this proc anymore.

File: code/_globalvars/lists/names.dm [hunk 5]
```diff
@@ -12,6 +12,8 @@ GLOBAL_LIST_INIT(lizard_names_male, world.file2list("strings/names/lizard_male.t
 GLOBAL_LIST_INIT(lizard_names_female, world.file2list("strings/names/lizard_female.txt"))
 GLOBAL_LIST_INIT(clown_names, world.file2list("strings/names/clown.txt"))
 GLOBAL_LIST_INIT(mime_names, world.file2list("strings/names/mime.txt"))
+GLOBAL_LIST_INIT(religion_names, world.file2list("strings/names/religion.txt"))
+GLOBAL_LIST_INIT(religion_meme_names, world.file2list("strings/names/religion_meme.txt"))
```
Comment (Rohesie) [inline code/_globalvars/lists/names.dm:16]:
```suggestion
```
No need for this anymore.

**Pull request merged**